exonum/src/messages2/authorization.rs

```diff
@@ -0,0 +1,145 @@
+use failure::Error;
+use hex::{FromHex, ToHex};
+
+use crypto::{
+    self, hash, Hash, PublicKey, SecretKey, Signature, PUBLIC_KEY_LENGTH, SIGNATURE_LENGTH,
+};
+
+use super::EMPTY_SIGNED_MESSAGE_SIZE;
+
+/// Correct raw message that was deserialized and verified, from `UncheckedBuffer`;
+/// inner data should be formed according to the following layout:
+/// | Position | Stored data |
+/// | - - - - - - - -| - - - - - - |
+/// | `0..32`  | author's PublicKey     |
+/// | `32`     | message class          |
+/// | `33`     | message type           |
+/// | `34..N`  | Payload                |
+/// | `N..N+64`| Signature                |
+///
+///
+/// Every creation of `SignedMessage` lead to signature verification, or data signing procedure,
+/// which can slowdown your code. Beware `SignedMessage` message, this procedure is not free.
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, Ord, PartialOrd)]
+pub struct SignedMessage {
+    pub(in messages) raw: Vec<u8>,
+}
+
+impl SignedMessage {
+    pub(crate) fn new(
+        cls: u8,
```

Comment:
What is the `cls`? Maybe this means `class`?

Reply:
Agree with @alekseysidorov. We can use `class` cause it's not reserved keyword.